src/irmin-pack/unix/file_manager.ml

```diff
@@ -537,7 +543,14 @@ struct
         ~auto_flush_callback:cb
     in
 
-    (* Step 2. Update the control file *)
+    (* Step 2. Transfer remaining newies from old_suffix to new_suffix. *)
+    let old_end_offset = Suffix.end_offset t.suffix in
+    let ( - ) = Int63.sub in
+    let remaining = old_end_offset - last_copied_offset |> Int63.to_int in
+    transfer ~src:t.suffix ~dst:new_suffix ~off:last_copied_offset
+      ~len:remaining;
```

Comment:
I think that `File_manager.swap` should only swap, and not deal with newies. Can you move this up to ext?

Also, I think you're missing a flush on dst.

Reply:
I think that `Gc` should contain a method that does the transfer (something like `val transfer : src:Suffix.t -> dst:Io.t -> start_offset:int63 -> end_offset:int63 -> (unit, _) result`?)

We could then call it once or twice a the end of `Gc.run` (from the child process), and once just before swap (from the main process).

Reply:
> I think that `File_manager.swap` should only swap, and not deal with newies. Can you move this up to ext?

I tried this (9a97909d86e6aa5758fcb49d8d954d6b62945132), but I think its necessary to open the new_suffix twice this way (doing it all in the `Fm.swap`, the new_suffix was opened only once). Is this what you had in mind? 

> I think that Gc should contain a method that does the transfer (something like `val transfer : src:Suffix.t -> dst:Io.t -> start_offset:int63 -> end_offset:int63 -> (unit, _) result?`)

I added this function but it raises errors instead of returning them - this is because I wanted to reuse existing io functions and not allocate buffers at every read. If you prefer, I can add the missing io function that returns a result type and use that instead.